### DISCUSSION TITLE
Help for GenMeshCylinder()

### DISCUSSION THREAD

**Discussion by ZeroOut** [raysan5/raylib#2810] [Q&A]
Hi!
I'm using raylib as graphic tool and open dynamics engine for physics calculating.
It's all good for cubes and spheres physics simulation but cylinders or cones.
For example, sphere rotation follows the right direction, but cylinder rotate the wrong axis and the position is not set from the center of it's mesh.
![output0](https://user-images.githubusercontent.com/19900527/204025224-2672df51-0864-4634-a8dd-5aea67bf92ee.gif)
![output1](https://user-images.githubusercontent.com/19900527/204025812-d31ad461-0a89-4a3d-942a-6105c7bbdf51.gif)
How ever, I don't think it's a bug, I just want some help for creating a cylinder that can get the same physics states like cube or spheres. So I'm using a long cube as cylinder rendering.
![output2](https://user-images.githubusercontent.com/19900527/204027583-143ca45a-ee1a-4afa-ad9e-db262d3c8f7c.gif)
Anyone could help? plz!

**Comment by ZeroOut** [comment]
Found two solutions.
1. load cylinder model by .obj file with right orientation.
2. set offset rotation for ode geom, and add a ode body (virtual body), set it's position to offset location, set fixed joint for real body and new body. Then render cylinder state by getting virtual body.

**Comment by chriscamacho** [comment]
basically what I do as well, usually with a cylinder that is 1 in radius and 1 in height and scaled to size

See https://bedroomcoders.co.uk for some OpenDE stuff - vehicles, trimesh for terrain etc

**Reply by ZeroOut** [reply]
Thank you first, and I have read your code, it is the best way for now.
However, could I store the cylinder model to the hard code instead of load the .obj file?

**Reply by chriscamacho** [reply]
If you are really are heart broken about loading resources.  Your best bet is to look at how gen cylinder works and implement your own version, this should be trivial - remember you should only ever need one cylinder mesh, scale it as appropriate for each physics item.

To be honest though its a tiny file and in any case after development and before release you should probably be doing something like this anyhow https://bedroomcoders.co.uk/physfs-and-raylib/
